shuriken/include/shuriken/parser/Macho/macho_sections.h

```diff
@@ -0,0 +1,76 @@
+//--------------------------------------------------------------------*- C++ -*-
+// Shuriken-Analyzer: library for bytecode analysis.
+// @author Farenain <kunai.static.analysis@gmail.com>
+//
+// @file section.h
+// @brief MachoSections of a MACHO file represented by a structure
+
+#ifndef SHURIKENLIB_MACHO_SECTIONS_H
+#define SHURIKENLIB_MACHO_SECTIONS_H
+
+#include "shuriken/common/shurikenstream.h"
+
+#include <cstring>
+#include <iostream>
+#include <vector>
+#include <memory>
+#include <map>
+
+namespace shuriken::parser::macho {
+    class MachoSections {
+    public:
+        /// @brief Structure with the definition of the MACHO section
+        /// all these values are later used for parsing the section
+        /// from MACHO
+        struct section_t {
+            char sectname[16];          //! section type
+            char segname[16];           //! segment type
+            uint64_t addr;              //! starting position of the section
+            uint64_t size;              //! size of the section
+            uint32_t offset;            //! data's offset of the section
+            uint32_t align;             //! alignment in memory, power of 2
+            uint32_t reloff;            //! offset to the relocation entries for this section
+            uint32_t nreloc;            //! number of relocation entries for this section
+            uint32_t flags;             //! flags
+            uint32_t reserved1;         //! reserved space for additional info
+            uint32_t reserved2;         //! reserved space for additional info
+            uint32_t reserved3;         //! reserved space for additional info
+        };
+
+        using sections_t = std::vector<std::shared_ptr<section_t>>;
```

Comment:
Check for the `std::shared_ptr` maybe you can use a `std::unique_ptr` and then a `std::reference_wrapper`. The shared_ptr is used when you need different objects to have the ownership of that memory, in our parsers mostly the data is for reading. Check it out whenever you can!